pageserver/src/layered_repository/delta_layer.rs

```diff
@@ -420,6 +421,28 @@ impl DeltaLayer {
         }
     }
 
+    fn temp_path_for(
+        conf: &PageServerConf,
+        timelineid: ZTimelineId,
+        tenantid: ZTenantId,
+        key_start: Key,
+        lsn_range: &Range<Lsn>,
+    ) -> PathBuf {
+        let rand_string: String = rand::thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(8)
+            .map(char::from)
+            .collect();
+
+        conf.timeline_path(&timelineid, &tenantid).join(format!(
+            "{}-XXX__{:016X}-{:016X}.temp.{}",
```

Comment:
```suggestion
            "{}-XXX__{:016X}-{:016X}.{}.temp",
```

It's better to keep the `temp` suffix, so we know which are temp files.